query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -250,6 +255,30 @@ private String convertOperator2String(Operator operator) {
     }
   }
 
+  //called for selection columns
+  private String convertExpression2String(Expression expression, Params.Builder paramsBuilder,int selectionCounter) {
```

Comment:
I think we do not need this extra function that takes a selection map. Using `this.functionConverter.convert` which handles avg_rate will already convert it to SUM. Requires ColumnIndex can pre-processed from the allSelectionList.